File: Leetcode /LC 021VV Series/2149. Rearrange Array Elements by Sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;

        int[] pos = new int[n/2];
        int[] neg = new int[n/2];

        int posIndex = 0;
        int negIndex = 0;

        for(int i=0;i<n;i++)
        {
            if(nums[i]>0) {
                pos[posIndex] = nums[i];
                posIndex++;
            }
            
            else {
                neg[negIndex] = nums[i];
                negIndex++;
            }
        }

        for(int i=0;i<n/2;i++)
        {
            nums[i*2] = pos[i];
            nums[i*2+1] = neg[i];
        }

        return nums;
    }
}

File: Leetcode /LC 021VV Series/2149. Rearrange Array Elements by Sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size();

        vector<int> pos, neg, ans;

        for(int i=0;i<n;i++)
        {
            if(nums[i]<0) neg.push_back(nums[i]);
            else pos.push_back(nums[i]);
        }

        for(int i=0;i<n/2;i++)
        {
            ans.push_back(pos[i]);
            ans.push_back(neg[i]);
        }

        return ans;


        // while(even < n && odd < n)
        // {
        //     if(nums[even]<0) {
        //         even++;
        //     }
        //     else 
        //     {
        //         if(nums[odd]>0) {
        //             odd++;
        //         }
        //         else 
        //         {
        //             swap(nums[even], nums[odd]);
        //             even++;
        //             odd++;
        //         }
        //     }
        // }
        // return nums;
    }
};
